ci/h2spec.sh

```diff
@@ -0,0 +1,129 @@
+#!/bin/bash
+set -eu
+set -o pipefail
+shopt -s extglob
+
+SEP="$(tput bold)⣷$(tput sgr0)"
+LOGFILE=$(mktemp -t l5d-h2spec.log)
+SPIN='⣾⣽⣻⢿⡿⣟⣯⣷'
+
+spin() {
+    local pid=$!
+    local i=0
+    while [ "$(ps a | awk '{print $1}' | grep $pid)" ]; do
+        i=$(( (i+1) %8 ))
+        printf "\r$(tput bold)${SPIN:$i:1}$(tput sgr0)"
+        sleep .1
+    done
+}
+
+echo "${SEP} running h2spec against linkerd..."
+# test if we have a h2spec binary - either in the working dir,
+# or on the PATH
+set +e
+WHICH_H2SPEC=$(which h2spec)
+if [[ $? -eq 0 ]]; then
+    # if which found an h2spec executable, just return that path.
+    H2SPEC_EXEC="${WHICH_H2SPEC}"
+    set -e
+else
+    set -e
+    # which didn't find a h2spec on the path. check if it exists in
+    # the working dir, and if not, dow nload it.
+    if [ -e "h2spec" ] ; then
+        # h2spec was found in the working dir, use that.
+        H2SPEC_EXEC="./h2spec"
+    else
+        # if we don't already have a h2spec executable, download it.
+        echo "${SEP} h2spec not found..."
+        # detect OS for downloading h2spec.
+        case $OSTYPE in
+            darwin*)
+              echo "${SEP} detected OS as macOS"
+              H2SPEC_TAR="h2spec_darwin_amd64.tar.gz"
+              ;;
+            linux*)
+              echo "${SEP} detected OS as linux"
+              H2SPEC_TAR="h2spec_linux_amd64.tar.ghz"
+              ;;
+        esac
+        # cd into /usr/local/bin to install h2spec.
+        DIR=$(pwd)
+        cd /usr/local/bin
```

Comment:
recommend putting `h2spec` into `~/bin`, as it's less invasive, and we may not have perms to for `/usr/local/bin`.

Reply:
is `~/bin` always on the path?

Reply:
it's not necessarily, but if you put it there, you can at least reference it directly.

Reply:
hmm – and forgive me if i'm entering into bikeshed territory here – it seems to me that if `~/bin` isn't necessarily on the path, we'll need to either add it to the path, or add a special case to check whether `h2spec` is installed in that location?

Reply:
sure, i don't think we want to modify the path, but checking for that install location seems reasonable.

Reply:
yeah, i didn't think we'd want to have the script modify the path either, but seems like checking if it exists in `~/bin` complicates this logic a bit... happy to add that though

Reply:
no more than checking for `./h2spec` ;)